### PR TITLE
Add image dependency support

### PR DESCRIPTION
**Pull request by jrrbru**
dust-loader-complete will now find  tags and resolve the images, adding them to webpack's dependency tree. It then generates a template function on the fly to use to insert the final webpack path back into the parent template once the file is loaded in the browser.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -82,122 +88,157 @@ function loader( source ) {
...
+      reg.lastIndex += (replace.length - result[0].length);
     }
   }
 
   return source;
 }
 
+// Look for <img> tags and require the actual files
+function findImages(templateName, source, deps, options) {
+  var reg = /(<img\s.+?src=['"])([^'"]+)(['"][^/>]+(?:\s+)?\/?>)/g,
```
Comment (jas7457) [inline index.js]: I don't think this regex works for the most basic case of `` Also, could we make it a bit more lenient and add optional spaces around the = in src=['"]

File: index.js [hunk 2]
```diff
@@ -82,122 +88,157 @@ function loader( source ) {
...
+      function body_0(chk, ctx) {
+          return chk.write(path);
+      }
+      return body_0;
+    })();`
+    deps.push(srcTemplate);
+
+    // replace the original image path with a partial template call to our method that will return the path.
+    const replace = `${result[1]}{>"${imageTemplateName}"/}${result[3]}`
+    source = source.substring(0, result.index) + replace + source.substring(result.index + result[0].length);
```
Comment (jas7457) [inline index.js]: Couldn't you just do something like:
`source = source.replace(result[0], replace)`

File: index.js (hunk 1 above)
Comment (jrrbru) [reply index.js]: I'll take a look

File: index.js (hunk 2 above)
Comment (jrrbru) [reply index.js]: Yeah, I think that's simpler. It does do one extra search for the information, but I think that's probably okay.

**Comment by jrrbru** [comment]
Updated based on review

File: index.js [hunk 3]
```diff
@@ -82,122 +88,157 @@ function loader( source ) {
...
+      reg.lastIndex += (replace.length - result[0].length);
     }
   }
 
   return source;
 }
 
+// Look for <img> tags and require the actual files
+function findImages(templateName, source, deps, options) {
+  var reg = /(<img\s.*src\s?=\s?['"])([^'"]+)(['"][^/>]*\/?>)/g,
```
Comment (jas7457) [inline index.js]: You should add a ? after img\s. or else if you have multiple img tags on the same line, it won't grab what you actually want. See https://regexr.com/3tb4m for an example

File: test/fixtures/image.dust [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{>"layouts/master" title="Image" /}
+
+{<bodyContent}
+	<img width="250px" src="./images/dolphin.jpg" style="display:block" />
+	<img src = "./images/dolphin.jpg"/>
+	<img src = "./images/dolphin.jpg"/>
```
Comment (jas7457) [inline test/fixtures/image.dust]: Did you mean to add the same image the same way twice?

File: index.js (hunk 3 above)
Comment (jrrbru) [reply index.js]: Fixed

File: test/fixtures/image.dust (hunk 4 above)
Comment (jrrbru) [reply test/fixtures/image.dust]: Yes, as a test. But I've changed it now to test multiple images on the same line

**Comment by jas7457** [comment]
Looks good. Will merge shortly

**Pull request merged**